src/gas/mod.rs

```diff
@@ -501,6 +502,20 @@ pub fn inject_gas_counter<R: Rules>(
 			elements::Section::Start(start_idx) => {
 				if *start_idx >= gas_func { *start_idx += 1}
 			},
+            elements::Section::Name(section) => {
+                if let Some(funcs) = section.functions_mut() {
+                    let mut new_names = elements::IndexMap::<String>::default();
+                    for (idx, func) in funcs.names().iter() {
+                        if idx > gas_func {
```

Comment:
Should this be a not strict comparison?

```suggestion
                        if idx >= gas_func {
```